data_steward/gcloud/gcs/__init__.py

```diff
@@ -151,5 +156,28 @@ def list_sub_prefixes(self, bucket_name: str, prefix: str) -> list:
         return list(pg_iterator)
 
     def _get_hpo_bucket_id(self, hpo_id: str) -> str:
-        # TODO reconsider how to map bucket name
-        return os.environ.get(f'BUCKET_NAME_{hpo_id.upper()}')
\ No newline at end of file
+        """
+        Get the name of an HPO site's private bucket
+        Empty/unset bucket indicates that the bucket is intentionally left blank and can be ignored
+        :param hpo_id: id of the HPO site
+        :return: name of the bucket
+        """
+        service = os.environ.get('GAE_SERVICE', 'default')
+
+        hpo_bucket_query = JINJA_ENV.from_string(SELECT_ALL_QUERY).render(
+            project_id=self.project,
+            dataset_id=LOOKUP_TABLES_DATASET_ID,
+            table_id=HPO_ID_BUCKET_NAME_TABLE_ID)
+
+        result_df: DataFrame = query(hpo_bucket_query)
+        condition_hpo_id = (result_df['hpo_id'] == hpo_id)
+        condition_service = (result_df['service'] == service)
+        result_filtered = result_df[condition_hpo_id & condition_service]
+
+        if result_filtered['bucket_name'].count() != 1:
+            raise BucketNotSet(
```

Comment:
Return None, 'None' or something instead. BucketNotset is used in the parent function